shell.nix

```diff
@@ -19,6 +19,7 @@ pkgs.mkShell {
 
     # Rust toolchain
     rustToolchain
+    rustup
```

Comment:
This change seems unrelated to the rest of the PR

Reply:
Yikes. This was not meant to be part of the PR